### PR TITLE
Api fix: split logic and GUI for view

### PR DESCRIPTION
**Pull request by Hpn4**
Added function in the `GUI` file for:
- Raw view
- Filter2D view
- Cuts view
- Lens view
- Some overlay
- browse file utility

Some logic have been changed:
- Change how accumulation queue for cuts are allocated/deallocated. This is now only done on `TimeTransformationCuts` and `DeleteTimeTransformationCuts` instead of each pipe refresh

### CODE REVIEW

File: Holovibes/sources/gui/windows/panels/view_panel.cc [hunk 1]
```diff
@@ -245,179 +222,63 @@ void ViewPanel::set_view_mode(const QString& value) { parent_->set_view_image_ty
...
-    // FIXME: if slice are closed, cancel time should be call.
-    else
+    if (api::set_3d_cuts_view(checked))
     {
-        cancel_time_transformation_cuts();
-        api::set_yz_enabled(false);
-        api::set_xz_enabled(false);
+        const ushort nImg = std::min(api::get_time_transformation_size(), time_transformation_cuts_window_max_size);
+        gui::set_3d_cuts_view(checked, 0);
+        parent_->notify(); // Make the x and y parameters visible
```
Comment (Baschtien) [inline Holovibes/sources/gui/windows/panels/view_panel.cc:235]: `nImg` doesn't seem to be used here... Maybe it should be in place of the 0 ?

**Review by Baschtien — COMMENTED** [review]
Ima be honest, I didn't understand a lot of what you did, especially what you deleted; but by judging how you tested and how most changes are just moving code, it looks good!

However, a lot of (mostly api) functions that take a boolean input have that input named `checked` instead of something more explicit for the context, like `enabled` for example. Is this following a convention ?

File: Holovibes/sources/API.cc [hunk 2]
```diff
@@ -725,16 +561,25 @@ void set_q_accu_level(uint value)
...
 void set_p_index(uint value)
 {
+    if (get_compute_mode() == Computation::Raw)
+        return;
+
+    if (value >= get_time_transformation_size() || value == 0)
+    {
+        LOG_ERROR("p param has to be between 1 and #img");
+        return;
+    }
```
[the remark this answers is not in the document]
Comment (Hpn4) [reply Holovibes/sources/API.cc:579]: It was already here, maybe a physical reason for skipping the first frequency. Ask michael if needed

**Review by Baschtien — APPROVED** [review]

File: Holovibes/includes/display_queue.hh [hunk 3]
```diff
@@ -20,6 +20,10 @@ class DisplayQueue
     {
     }
 
+    /*! Return the last image in the Queue of nullptr if the Queue is empty
+     *
+     * \return The last image of size fd.frame_res or nullptr if no frame
+     */
```
Comment (simon-riou) [inline Holovibes/includes/display_queue.hh:26]: thanks for adding comments

File: Holovibes/sources/gui/windows/panels/import_panel.cc [hunk 4]
```diff
@@ -132,17 +137,17 @@ void ImportPanel::import_file(const QString& filename)
...
         // Gather data from the newly opened file
-        size_t nb_frames = input_file->get_total_nb_frames();
+        int nb_frames = static_cast<int>(input_file->get_total_nb_frames());
         UserInterfaceDescriptor::instance().file_fd_ = input_file->get_frame_descriptor();
 
         // Don't need the input file anymore
         delete input_file;
 
         // Update the ui with the gathered data
         // The start index cannot exceed the end index
```
Comment (simon-riou) [inline Holovibes/sources/gui/windows/panels/import_panel.cc:147]: Good job

File: Holovibes/includes/API.hh [hunk 5]
```diff
@@ -277,7 +262,8 @@ void enable_pipe_refresh();
  */
 void disable_pipe_refresh();
 
-void create_holo_window(ushort window_size);
+void create_window(Computation window_kind, ushort window_size);
+
```
Comment (ArthurCourselle) [inline Holovibes/includes/API.hh:266]: Can you add some doc please ?

**Review by simon-riou — APPROVED** [review]

File: Holovibes/includes/gui/windows/AdvancedSettingsWindow.hh [hunk 6]
```diff
@@ -39,6 +39,12 @@ class AdvancedSettingsWindow : public QMainWindow
     /*! \brief Destroy the AdvancedSettingsWindow object. */
     ~AdvancedSettingsWindow();
 
+    /*! \brief Set the callback function called when user click on the Save button
+     *
+     * \param callback the function to call
```
Comment (ArthurCourselle) [inline Holovibes/includes/gui/windows/AdvancedSettingsWindow.hh:44]: Please precise [in] or [out]

File: Holovibes/sources/gui/GUI.cc [hunk 7]
```diff
@@ -1,17 +1,187 @@
...
+}
+
+void set_chart_display(bool enabled)
+{
+    if (enabled)
+    {
+        UI.plot_window_.reset(new gui::PlotWindow(*api::get_compute_pipe()->get_chart_display_queue().get(),
+                                                  UI.auto_scale_point_threshold_,
+                                                  "Chart"));
+    }
```
Comment (ArthurCourselle) [inline Holovibes/sources/gui/GUI.cc:89]: 1 line if should not have brackets.

File: Holovibes/sources/gui/GUI.cc [hunk 8]
```diff
@@ -1,17 +1,187 @@
...
+    UI.advanced_settings_window_->set_callback(callback);
+}
+
+/**
+ * @brief Extract the name from the filename
+ *
+ * @param filePath the file name
+ * @return std::string the name extracted from the filename
+ */
+std::string getNameFromFilename(const std::string& filename)
```
Comment (ArthurCourselle) [inline Holovibes/sources/gui/GUI.cc:159]: Please do like the other doxygen comments in holovibes, use `\` and not `@`

File: Holovibes/sources/gui/windows/panels/export_panel.cc [hunk 9]
```diff
@@ -42,6 +36,14 @@ void actualise_record_output_file_ui(const std::filesystem::path file_path)
...
+{
+    ui_->NumberOfFramesSpinBox->setSingleStep(record_frame_step_);
+    set_record_mode(static_cast<int>(RecordMode::RAW)); // Not great but it works
+
+    actualise_record_output_file_ui(std::filesystem::path(ui_->OutputFilePathLineEdit->text().toStdString()));
+}
+
 void ExportPanel::on_notify()
 {
     // TODO
```
Comment (ArthurCourselle) [inline Holovibes/sources/gui/windows/panels/export_panel.cc:49]: What is this TODO ? If not yours or you don't know it's origin please remove it.

**Review by ArthurCourselle — CHANGES_REQUESTED** [review]
Great, I just requested 2-3 changes but only documentation purpose.

**Review by ArthurCourselle — CHANGES_REQUESTED** [review]

**Pull request merged**